src/Components/Server/src/Endpoints/CircuitMatcherPolicy.cs

```diff
@@ -0,0 +1,232 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Server.Circuits;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Matching;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    internal class CircuitMatcherPolicy : MatcherPolicy, IEndpointSelectorPolicy
+    {
+        internal const string IdClaimType = "bcid";
+        internal const string CookiePrefix = "Circuit.";
+
+        private const int PrefixLenght = 4;
```

Comment:
```suggestion
        private const int PrefixLength = 4;
```

Reply:
Also, are we relying on non-collision of the IDs up to this prefix length for a given user? If so we might be better adding another couple of characters to the prefix length.

(e.g., if these are base64 chars, the 4-char prefix space size is 64^4=16 million)

Reply:
Assuming you have 50 open circuits on a client at the same time, `0.00000298023223876953125‬` those are your chances of running into a collision I believe. We can add a couple more and won't hurt, but I think its already within target.

Reply:
Sure. If each user has 2 circuits, then the expected number of users until you get a clash is about 12 million. It is very unlikely for any given user, but people do win lotteries :)